ccmlib/node.py

```diff
@@ -1254,22 +1257,34 @@ def _update_pid(self, process):
             raise NodeError('Problem starting node %s' % self.name, process)
         self.__update_status()
 
-    def __gather_sstables(self, datafile=None, keyspace=None, columnfamilies=None):
-        datafiles = []
+    def __gather_sstables(self, datafiles=None, keyspace=None, columnfamilies=None):
+        files = []
         if keyspace is None:
-            for k in self.list_keyspaces():
-                datafiles = datafiles + self.get_sstables(k, "")
-        elif datafile is None:
+            if datafiles:
+                for datafile in datafiles:
+                    if not os.path.isabs(datafile):
+                        datafile = os.path.join(os.getcwd(), datafile)
+
+                    if not os.path.exists(datafile):
+                        print_("File doesn't exist: " + datafile)
+                    elif datafile.endswith('Data.db') or datafile.endswith('big-Data.db'):
+                        files.append(datafile)
+            else:
+                for k in self.list_keyspaces():
+                    files = files + self.get_sstables(k, "")
+        elif datafiles is None:
             if columnfamilies is None:
-                datafiles = datafiles + self.get_sstables(keyspace, "")
+                files = files + self.get_sstables(keyspace, "")
             else:
                 for cf in columnfamilies:
-                    datafiles = datafiles + self.get_sstables(keyspace, cf)
+                    files = files + self.get_sstables(keyspace, cf)
         else:
+            # What is this used for? I don't see where it could be useful... it concatenates the
+            # keyspace dir with a sstable file? Can we remove?
```

Comment:
Can you remove this comment? This is used for when the user passes in the keyspace name, the CF name, and the sstable name. It does appear to be broken, as it is not considering the fact that the CF should be in the path. If you could fix this as part of this PR, that would be appreciated.